File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/TypeEntity.java
package com.github.ol_loginov.heaplibweb.repository.heap;

import com.github.ol_loginov.heaplibweb.repository.EntityIdentity;
import lombok.*;

@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Getter
@Setter
public class TypeEntity extends EntityIdentity {
	private int heapId;
	private String name;
}

File: heap-web/src/main/java/com/github/ol_loginov/heaplibweb/repository/heap/TypeEntity.kt
package com.github.ol_loginov.heaplibweb.repository.heap

import com.github.ol_loginov.heaplibweb.repository.EntityIdentity

class TypeEntity private constructor(
    id: Int,
    var name: String
) : EntityIdentity(id) {
    internal constructor() : this(0, "")
    constructor(name: String) : this(0, name)
}